### ISSUE TITLE
Shown Tax ID in quotation belongs to other customer, not the one quotation is addressed to

### ISSUE THREAD

**Issue by jaimestuardo** [frappe/erpnext#23251]
I have this problem actually in quotation print format, but as I have seen when searching in Google, it also happens in other doc types.

My ERP Version is 

**ERPNext: v12.11.2 (version-12)

Frappe Framework: v12.9.1 (version-12)**

But this problem existed when I had version 11 as well.

The Jinja instruction is:

    {{ frappe.db.get_value("Customer", doc.customer, "tax_id") or "" }}

Since tax ID appears empty in quotation. I have checked tabCustomer table and there the tax id is correctly set for the customer addressed of the quotation.

I have seen in tabCustomer table also a field named `email_id` so I did a test by changing above instruction to:

    {{ frappe.db.get_value("Customer", doc.customer, "email_id") or "" }}

My surprise was that the displayed e-mail belongs to other customer. Then I edited that other customer and set it own tax id. Then I changed the jinja instruction to use `tax_id` again and after displayed the quotation for a customer different from that, the tax id was the one that I have changed.

In summary, 

    {{ frappe.db.get_value("Customer", doc.customer, "tax_id") or "" }}

displays the tax id belonging to other customer. Not the one the quotation is addressed to.

**Comment by jaimestuardo** [comment]
I have discovered something interesting.

I have saved other customer so that he appears at the top of the customers list. Then I loaded the customer's quotation, and surprise, the TAX ID that appeared belongs to the updated customer and not the customer the quotation is addressed to.

**Comment by jaimestuardo** [comment]
At last! After a lot of attempts I could do it.

I used finally this instruction:

    {{ frappe.db.get_value("Customer", {'name': doc.customer_name}, "tax_id") or "" }}

Due to a strange reason, system could not link both Quotation and Customer tables. One of the  attempts was to add Customer link in `Quotation` DocType but it did no work.

My guessing is that since `Quotation` table has already a field named `customer`, it could not link that with the Customer table and the `name` field.

By comparing `contact_person` field of `Quotation` DocType, I don't really know why that link to `Contact` table does work and the link to `Customer` table does not.

This is clearly a bug from my point of view but the important thing now is that the workaround worked 😄 

I will keep this issue opened because I think the link between `Quotation` and `Customer` tables should be fixed.

Regards
Jaime

**Comment by deepeshgarg007** [comment]
@jaimestuardo This looks like some customization issue and not a bug, closing this issue
